model/task/task.go

```diff
@@ -2503,10 +2501,23 @@ func (t *Task) Archive() error {
 			"$inc": bson.M{ExecutionKey: 1},
 		})
 	if err != nil {
-		return errors.Wrap(err, "updating task")
+		return false, errors.Wrap(err, "updating task")
+	}
+	if info.Updated <= 0 {
```

Comment:
I see what you mean. I may be misinterpreting but `Archived = true` does not currently happen on the task during archived, it just gets set on the archived task that is sent to the old tasks collection. However I've now made the change to set `Archived` in the DB on the task doc during archiving operation and adding that to the query done before we reset the tasks. So that should make it so that if the reset task functions fail between archiving and resetting, we can retry the operation and it'll succeed.

Reply:
Ah yeah, I didn't notice that archive is only set on the old task. Your change looks good though!